test/performance/performance.go

```diff
@@ -0,0 +1,389 @@
+	if err != nil {
+		klog.Exitf("Error getting thanos-querier route: %s", err)
+	}
+	var thanos_route map[string]interface{}
+	if err := json.Unmarshal(route_output, &thanos_route); err != nil {
+		klog.Exitf("Error unmarshalling route thanos-querier to JSON", err)
+	}
+	thanos_host := thanos_route["spec"].(map[string]interface{})["host"].(string)
+	if err != nil {
+		klog.Exitf("Error decoding thanos host from route thanos-querier", err)
+	}
+
+	klog.Info("Setup complete! Starting the Config Policy Controller performance test :)")
+	table_data := []metricData{}
+
+	table_data = append(table_data, getMetrics(thanos_host, string(token), per_batch_sleep, 0, insecure))
+
+	//setup temp directory for auto-generated policy YAML to live in
+	policy_dir, err := os.MkdirTemp(path.Join(performance_dir, "resources"), "policies")
+	if err != nil {
+		klog.Exitf("Error creating temp policy directory: %s", err)
+	}
+	defer os.RemoveAll(policy_dir) // clean up
+
+	total_plcs := 0
+	for total_plcs < n_total {
+		start := time.Now()
+
+		// create a batch of policies
+		klog.V(2).Infof("Creating %d copies of %s on the managed cluster...\n", n_per_batch, plc_filename)
+		batch_fails := 0
+		for batch_plcs := 0; batch_plcs < n_per_batch; batch_plcs += 1 {
+			err = genUniquePolicy(path.Join(performance_dir, plc_filename), path.Join(policy_dir, "current_policy.yaml"))
+			if err != nil {
+				klog.Exitf("Error patching policy with unique name: %s", err)
+			}
+
+			tries := 2
+			for tries > 0 {
+				creation_output, err := exec.Command("kubectl", "apply", "-f", path.Join(policy_dir, "current_policy.yaml")).CombinedOutput()
+				if err != nil {
+					klog.Error("Error creating policy: %s, %s", err, string(creation_output))
+					tries -= 1
+					if tries == 0 {
+						os.Exit(1)
+					}
+				} else {
+					break
+				}
+			}
+			total_plcs += 1
+		}
+
+		// sleep for the remainder of per_batch_sleep
+		end := time.Now()
+		elapsed := end.Sub(start)
+		bonus_time := time.Duration(per_batch_sleep)*time.Minute - elapsed
+
+		klog.V(2).Infof(
+			"%d policies created in %.2f seconds! Waiting an additional %.2f seconds for policies to process...\n",
+			n_per_batch-batch_fails,
+			elapsed.Seconds(),
+			bonus_time.Seconds(),
+		)
+
+		time.Sleep(bonus_time)
+
+		table_data = append(table_data, getMetrics(thanos_host, string(token), per_batch_sleep, total_plcs, insecure))
+	}
+
+	printTable(table_data)
+
+	wd, err := os.Getwd()
+	if err != nil {
+		klog.Errorf("Error getting working directory: %s", err)
+	}
+	outpath := path.Join(wd, performance_dir, "output")
+	err = os.MkdirAll(outpath, os.ModePerm)
+	if err != nil {
+		klog.Errorf("Error creating output directory: %s", err)
+	}
+	exportTable(table_data, path.Join(performance_dir, "output", output_filename))
+
+	klog.Info("Performance test completed! Cleaning up...")
+	yaml_cleanup, err := exec.Command("rm", path.Join(policy_dir, "current_policy.yaml")).CombinedOutput()
```

Comment:
This isn't needed anymore because of your `defer` that cleans up all of `policy_dir`.